### ISSUE TITLE
Consider to add a new error code for `urVirtualMemMap` when the VA is already mapped

### ISSUE THREAD

**Issue by AllanZyne** [oneapi-src/unified-runtime#1174]
Currently, the error code of `urVirtualMemMap` is invalid argument when the VA is already mapped. 
But this error code is not clear enough, since returning invalid argument could also be due to other reasons, like the `size` is not proper.

**Comment by pbalcer** [comment]
For the [sanitizer](https://github.com/oneapi-src/unified-runtime/pull/1074/files#diff-cdb245c314b0deaa1e8277e335e881a8f7985456bb01ebcfe92aa4321b1de6ceR336) we'd ideally have a dedicated error code for VA already having a physical mapping, because in that case it essentially means "success". Using a generic error code like invalid operation might lead to it being reused later on for something different.

**Comment by AllanZyne** [comment]
> For the [sanitizer](https://github.com/oneapi-src/unified-runtime/pull/1074/files#diff-cdb245c314b0deaa1e8277e335e881a8f7985456bb01ebcfe92aa4321b1de6ceR336) we'd ideally have a dedicated error code for VA already having a physical mapping, because in that case it essentially means "success". Using a generic error code like invalid operation might lead to it being reused later on for something different.

Yes, I totally agree. But if we can document that "UR_RESULT_ERROR_INVALID_OPERATION" only means the mapping failed, I think using the existing error code is fine as well.

**Comment by alycm** [comment]
@steffenlarsen Do you have thoughts on this?  Is a new error code for this situation okay, and does it seem like all adapters could accurately provide it?

**Comment by AllanZyne** [comment]
> Sadly I don't see the [cuMemMap](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__VA.html#group__CUDA__VA_1gff1d395423af5c5c75375516959dae56) function specifying a special error code for that case, so I don't know if it reports an error in the mentioned scenario and if it does, whether that error is distinct like what is asked for here.

I'm not sure, just from the document, it seems like "CUDA_ERROR_OUT_OF_MEMORY" is what I want.
